arviz/plots/backends/matplotlib/traceplot.py

```diff
@@ -127,8 +127,10 @@ def plot_trace(
 
     # Check the input for lines
     if lines is not None:
-        if lines[0] not in [plotters[i][0] for i in range(len(plotters))]:
-            warnings.warn("A valid var_name should be provided")
+        all_var_names = set([plotters[i][0] for i in range(len(plotters))])
+        for line in lines:
+            if line[0] not in all_var_names:
+                warnings.warn("A valid var_name should be provided, found {} expected one of {}".format(line[0], all_var_names))
```

Comment:
This will print one warning per missing variable, we could try to tell about all of them at once:

```
missing_vars = set([line for line in lines if line not in all_var_names])
warnings.warn("{} var_names were not found in data to be plotted ({}) and will be omitted".format(missing_vars, all_var_names))
```

Please improve and adapt on this if you like the idea